tests/end_to_end/conftest.py

```diff
@@ -36,6 +38,24 @@ def pytest_addoption(parser):
     parser.addoption("--log_memory_usage", action="store_true")
 
 
+def pytest_configure(config):
+    """
+    Configure the pytest plugin.
+    Args:
+        config: pytest config object
+    """
+    # Declare some global variables
+    args = parse_arguments()
+    # Use the model name from the test case name if not provided as a command line argument
+    config.model_name = args.model_name
+    config.num_collaborators = args.num_collaborators
+    config.num_rounds = args.num_rounds
+    config.require_client_auth = not args.disable_client_auth
+    config.use_tls = not args.disable_tls
+    config.log_memory_usage = args.log_memory_usage
```

Comment:
Also consider #1195 changes for memory usage logging.

Reply:
Sure. We will take it up in subsequent PR.